src/wallet/wallet.cpp

```diff
@@ -558,6 +558,106 @@ void CWallet::UpgradeDescriptorCache()
     SetWalletFlag(WALLET_FLAG_LAST_HARDENED_XPUB_CACHED);
 }
 
+/**
```

Comment:
ea7a61ca94cc0d59cfe25307cd60fcda2aedec5f nit, keeping it closer to 80 characters wide, and drops "Any other candidates will be added as HD keys that have been rotated out.":

```
/**
 * UpgradeToGlobalHDKey searches through the descriptors in a descriptor wallet
 * and tries to find a CExtPubKey which is likely to be the most recent one used
 * to generate all of the automatically generated descriptors. The automatically
 * generated descriptors are guaranteed to be 2 p2pkh, 2 p2sh-segwit, and 2 p2wpkh.
 * There may be 2 p2tr descriptors, as well as imports. Those are ignored as they
 * are not guaranteed to be present in all wallets.
 * Candidate CExtKeys are those that appear in the aforementioned descriptors as
 * the only key. The best candidate is selected as the one that was most recently
 * used.
 *
 * The best candidate will be added to the wallet as the current active HD key.
 *
 * WALLET_FLAG_GLOBAL_HD_KEY will be set in order to indicate that this upgrade
 * has occurred, regardless of whether the active HD key was found.
 */
```

I appended ", regardless…"